src/PlottingInterface.cc

```diff
@@ -437,50 +556,30 @@ void PlottingInterface::InitTimer()
 {
   this->dataPtr->timer = new QTimer();
   this->dataPtr->timer->setInterval(this->dataPtr->timeout);
-  connect(this->dataPtr->timer, SIGNAL(timeout()), this, SLOT(UpdateGui()));
+  connect(this->dataPtr->timer, SIGNAL(timeout()), this, SLOT(UpdateTime()));
   this->dataPtr->timer->start();
-
-  auto moveTimer = new QTimer();
-  moveTimer->setInterval(1000);
-  connect(moveTimer, SIGNAL(timeout()), this, SLOT(moveCharts()));
 }
 
-////////////////////////////////////////////
-void PlottingInterface::UpdateGui()
+//////////////////////////////////////////////////////
+void PlottingInterface::onPlot(int _chart, QString _fieldID, double _x, double _y)
 {
-  this->dataPtr->transport->UnsubscribeOutdatedTopics();
-
-  auto topics = this->dataPtr->transport->Topics();
+  if (int(_x) == DEFAULT_TIME)
```

Comment:
By casting `double` to `int`, we are basically rounding down the digits. Will `_x` always be a integer number or is it possible for `_x` to be rounded down from `2.5` to `2`? Maybe adding a comment to explain the purpose of this type casting?

Reply:
this is just to prevent a compiler error of comparing the _x (which is time which is double) to a default integer number 
i use this as a flag that lets me know whether it has a header or not

but i suppose that the header time will never be negative ... is that ok ? or users may have a -ne header time ?